File: problems/1321-get-equal-substrings-within-budget/solution.java
class Solution {
    public int equalSubstring(String s, String t, int maxCost) {
        int n = s.length();
        int curr_cost =0 ;
        int max_length = 0;
        int start = 0;

        for(int i=0; i<n;i++) {
            curr_cost += Math.abs((int)s.charAt(i) - (int)t.charAt(i));

            while(curr_cost > maxCost){
                curr_cost -= Math.abs((int)s.charAt(start) - (int)t.charAt(start));
                start++;
            }

            max_length = Math.max(max_length, i-start+1);
        }
        return max_length;
    }
}

File: problems/1321-get-equal-substrings-within-budget/solution.py
class Solution(object):
    def equalSubstring(self, s, t, maxCost):
        n = len(s)
        start = 0
        curr_cost = 0
        max_len = 0
        # ord represents unicode :: ASCII in this case
        for end in range(n):
            curr_cost += abs(ord(s[end]) - ord(t[end]))

            while curr_cost > maxCost:
                curr_cost -= abs(ord(s[start]) - ord(t[start]))
                start+=1
            
            max_len = max(max_len, end - start+1)
        
        return max_len
        
